File: src/fable-library-rust/Option.rs
pub mod Option {
    use std::rc::Rc;

    pub fn getValue<T: Clone>(opt: &Rc<Option<T>>) -> T {
        match (*opt).as_ref() {
            Some(x) => x.clone(),
            None => panic!("Option has no value")
        }
    }

    pub fn bind<T: Clone, U: Clone>(binder: &impl Fn(&T) -> Rc<Option<U>>, opt: &Rc<Option<T>>) -> Rc<Option<U>> {
        match (*opt).as_ref() {
            Some(x) => Rc::from(binder(x)),
            None => Rc::from(None)
        }
    }

    pub fn map<T: Clone, U: Clone>(mapping: &impl Fn(&T) -> U, opt: &Rc<Option<T>>) -> Rc<Option<U>> {
        match (*opt).as_ref() {
            Some(x) => Rc::from(Some(mapping(x))),
            None => Rc::from(None)
        }
    }
}

File: src/fable-library-rust/fable/Option.fs
module Option

// [<Fable.Core.Import("getValue", "Native.rs|Option")>]
// let getValue<'T> (opt: 'T option): 'T = Fable.Core.Util.nativeOnly

let getValue<'T> (opt: 'T option): 'T =
    match opt with
    | Some x -> x
    | None -> failwith "Option has no value"

let bind<'T, 'U> (binder: 'T -> 'U option) (opt: 'T option): 'U option =
    match opt with
    | Some x -> binder x
    | None -> None

let map<'T, 'U> (mapping: 'T -> 'U) (opt: 'T option): 'U option =
    match opt with
    | Some x -> Some (mapping x)
    | None -> None
